### PR TITLE
Update packaging setup

### PR DESCRIPTION
**Pull request by xhochy**
# Description:

Use declarative packaging, move to `environment.yml` and limit supported Python versions to 3.8-3.10.

- [ ] ~Closes #xxxx~
- [x] Changelog entry

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: plateau
+channels:
+  - conda-forge
+  - nodefaults
+dependencies:
+  - dask[dataframe]!=2021.5.1,!=2021.6.0  # gh475 - 2021.5.1 and 2021.6.0 broke ci, omit those versions
```
Comment (jtilly) [inline environment.yml:6]: Does the `[dataframe]` syntax do anything?

File: environment.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  # Currently dask and numpy==1.16.0 clash
+  # TODO: add support for numpy>=1.23
+  - numpy!=1.15.0,!=1.16.0
+  - pandas>=0.23.0, !=1.0.0
+  - pyarrow>=0.17.1,!=1.0.0, <10
+  - simplejson
+  - simplekv
+  - storefact
+  - toolz
+  - typing_extensions # Some backports of the py3.8 typing module
```
Comment (jtilly) [inline environment.yml:18]: Still necessary with `python>=3.8`? If so, then please fix the comment.

File: setup.cfg [hunk 3]
```diff
@@ -1,5 +1,38 @@
...
+    msgpack>=0.5.2
+    # Currently dask and numpy==1.16.0 clash
+    numpy!=1.15.0,!=1.16.0
+    pandas>=0.23.0, !=1.0.0
+    pyarrow>=0.17.1,!=1.0.0,<7
+    simplejson
+    simplekv
+    storefact
+    toolz
+    typing_extensions; python_version<"3.8"  # Some backports of the py3.8 typing module
```
Comment (jtilly) [inline setup.cfg:27]: Same as above.

**Review by jtilly — APPROVED** [review]

File: environment.yml (hunk 1 above)
Comment (xhochy) [reply environment.yml]: No, it doesn't.

**Comment by jtilly** [comment]
https://github.com/data-engineering-collective/plateau/blob/d5b17a2a156789d5ca91bcc106c4d96a3aaadb86/plateau/io_components/utils.py#L17-L20 is not needed anymore.

**Pull request merged**